peerinst/views/decorators.py

```diff
@@ -113,11 +107,10 @@ def wrapper(req, *args, **kwargs):
 
 
 def teacher_required(fct):
-    @login_required
     def wrapper(req, *args, **kwargs):
-        try:
-            teacher = Teacher.objects.get(user=req.user)
-        except Teacher.DoesNotExist:
+        if hasattr(req.user, "teacher"):
```

Comment:
Why the refactor?  In my opinion the original was clearer and is a bit more flexible as we can change the Manager behaviour.  (And hasattr is probably not a design pattern we want to make extensive use of for testing until we get to python3.)

Reply:
The pattern in the original `try` block is repeated in a lot of places, so it is probably worth the debate over which pattern we should use.  

So I thought about it a bit:  I may be wrong but doesn't the `hasattr` pattern require an extra query?

Reply:
You're right that while on python 2, we should stick to `try... except`.